selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/dependencies/OpenExternalLibraryFileAfterRefreshTest.java

```diff
@@ -0,0 +1,94 @@
+
+import com.google.inject.Inject;
+import java.net.URL;
+import java.nio.file.Paths;
+import org.eclipse.che.commons.lang.NameGenerator;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.client.TestProjectServiceClient;
+import org.eclipse.che.selenium.core.project.ProjectTemplates;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.CodenvyEditor;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class OpenExternalLibraryFileAfterRefreshTest {
+  private static final String PROJECT_NAME = NameGenerator.generate("ExternalFileTest", 4);
+  private static final String EXPECTED_EDITOR_TEXT =
+      "\n"
+          + " // Failed to get sources. Instead, stub sources have been generated.\n"
+          + " // Implementation of methods is unavailable.\n"
+          + "package javax.servlet;\n"
+          + "public interface Filter {\n"
+          + "\n"
+          + "    public void init(javax.servlet.FilterConfig arg0) throws javax.servlet.ServletException;\n"
+          + "\n"
+          + "    public void doFilter(javax.servlet.ServletRequest arg0, javax.servlet.ServletResponse arg1, javax.servlet.FilterChain arg2) throws java.io.IOException, javax.servlet.ServletException;\n"
+          + "\n"
+          + "    public void destroy();\n"
+          + "\n"
+          + "}\n";
+
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private TestWorkspace testWorkspace;
+  @Inject private Ide ide;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private CodenvyEditor editor;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    URL resource = getClass().getResource("/projects/default-spring-project");
+    testProjectServiceClient.importProject(
+        testWorkspace.getId(),
+        Paths.get(resource.toURI()),
+        PROJECT_NAME,
+        ProjectTemplates.MAVEN_SPRING);
+    ide.open(testWorkspace);
+  }
+
+  @Test
+  public void shouldOpenTheSameFileAfterRefreshBrowser() {
+    // prepare
+    projectExplorer.waitProjectExplorer();
+    projectExplorer.waitItem(PROJECT_NAME);
+
+    // open file from external library and check displayed text in editor
+    projectExplorer.openItemByPath(PROJECT_NAME);
+    openFileFromExternalLibraryAndCheckDisplayedText();
+
+    seleniumWebDriver.navigate().refresh();
+
+    // check that after browser refresh is still possible to open the same file
+    projectExplorer.waitProjectExplorer();
+    projectExplorer.waitItem(PROJECT_NAME);
+    openFileFromExternalLibraryAndCheckDisplayedText();
+  }
+
+  private void openFileFromExternalLibraryAndCheckDisplayedText() {
+    projectExplorer.waitVisibilityByName("README.md");
```

Comment:
We could shorten a code if we will have the method:
`projectExplorer.waitVisibilityByName(String... names)`
`projectExplorer.openItemByVisibleNameInExplorer(String... names)`

Reply:
The method "projectExplorer.openItemByVisibleNameInExplorer(String... names)" already has the "projectExplorer.waitVisibilityByName(String... names)" method inside. 
In this particular case checks node expanding, not displaying element before clicking.
```
projectExplorer.waitVisibilityByName("README.md");
projectExplorer.waitVisibilityByName("pom.xml");
projectExplorer.waitVisibilityByName("External Libraries");
```